src/dev/register_git_hook/register_git_hook.js

```diff
@@ -51,8 +52,48 @@ function getKbnPrecommitGitHookScript(rootPath) {
     command -v node >/dev/null 2>&1
   }
   
+  has_nvm() {
+    command -v nvm >/dev/null 2>&1
+  }
+  
+  try_load_node_from_nvm_paths () {
+    # If nvm is not loaded, load it
+    has_node || {
+      NVM_SH="${nodeHome}/.nvm/nvm.sh"
+      
+      if [ "${platform}" == "darwin" ] && [ -s "$(brew --prefix nvm)/nvm.sh" ]; then
+        NVM_SH="$(brew --prefix nvm)/nvm.sh"
+      fi
+      
+      export NVM_DIR=${nodeHome}/.nvm
+      
+      [ -s "$NVM_SH" ] && \. "$NVM_SH"
+      
+      # If nvm has been loaded correctly, use project .nvmrc
+      has_nvm && nvm use
+    }
+  }
+  
+  extend_user_path() {
+    if [ "${platform}" == "win32" ]; then
+      export PATH="$PATH:/c/Program Files/nodejs"
+    else 
+      export PATH="$PATH:/usr/local/bin:/usr/local"
+      try_load_node_from_nvm_paths
+    fi
+  }
+  
+  # Extend path with common path locations for node
+  # in order to make the hook working on git GUI apps
+  extend_user_path
+  
   # Check if we have node js bin in path
-  has_node || { echo "Can't found node bin in the PATH. Please update the PATH to proceed"; exit 1; }
+  has_node || { 
+    echo "Can't found node bin in the PATH. Please update the PATH to proceed."
+    echo "If your PATH already has the node bin, maybe you are using some git GUI app."
```

Comment:
I don't understand how an app would be executing this and it to not include the PATH. Is it explicitly setting the PATH? Or is it that nvm/n/etc is not defined for non-login shells?

Reply:
It happens a lot in GUI apps. I was not able to trace the root cause but every git-hooks library suffers from this problem. Just try for example that: install the git-tower, setup a git hook with an echo for the $PATH, and you will see the outputted $PATH in the Git-tower console will be different from the one you have in the bash. Same happens for example in [SourceTree ](https://community.atlassian.com/t5/Bitbucket-questions/SourceTree-Hook-failing-because-paths-don-t-seem-to-be-set/qaq-p/274792)

Reply:
Sounds like an issue for non-login shells then. Could we provide some guidance on what to fix? (Setup nvm for non-login shells or add node binary to non-login shell path)

Reply:
Basically the script is outputting that it cannot find node bin in the path. Don't u think this is responsibility of the people to just configure correctly their own local setup? I just added the auto guessing to try to help into the GUI apps as I'm considering the people using such kind of apps could not have the knowledge to lead with those problems. Husky was doing the same and we were dealing with it. What do you think guys?

Reply:
If it's working the same then that's probably fine, as most people will have figured this out already if they're using a GUI app. My only problem with the message is it makes it sound like it's unfixable. I'd suggest at least adding a note about adding node to the PATH for non-login shells.

Reply:
This is not related with the login or the non login shells. This is a problem related with the path config used by the application who are launching the GUI apps. For example in the macOS the sourcetree will be launched with `launchctl` and that application won't have the same path as $PATH. That's why try to auto guessing the path in real time will work, because it will already run inside the application launching the pre-commit hook. Maybe we can change the message of the script to:

```bash
echo "Can't found node bin in the PATH. Please update the PATH to proceed."
echo "If your PATH already has the node bin, maybe you are using some git GUI app not launched from the shell."
echo "In order to proceed, you need to config the PATH used by the application launching your git GUI app."
echo "If you are running macOS, you can do that using 'sudo launchctl config user path /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'"
```

What do you think @joshdover ?  You can found more info about the problem on https://community.atlassian.com/t5/Bitbucket-questions/SourceTree-Hook-failing-because-paths-don-t-seem-to-be-set/qaq-p/274792

Reply:
Done